extensions/notebook/src/test/common/utils.test.ts

```diff
@@ -134,7 +137,169 @@ describe('Utils Tests', function () {
+
+		it('runs successfully', async () => {
+			await utils.executeBufferedCommand('echo hello', {}, new MockOutputChannel());
+		});
+
+		it('errors correctly with invalid command', async () => {
+			await should(utils.executeBufferedCommand('invalidcommand', {}, new MockOutputChannel())).be.rejected();
+		});
+	});
+
+	describe('executeStreamedCommand', () => {
+
+		it('runs successfully', async () => {
+			await utils.executeStreamedCommand('echo hello', {}, new MockOutputChannel());
+		});
+
+		it('errors correctly with invalid command', async () => {
+			await should(utils.executeStreamedCommand('invalidcommand', {}, new MockOutputChannel())).be.rejected();
+		});
+	});
+
+	describe('isEditorTitleFree', () => {
+		it('title is free', () => {
+			should(utils.isEditorTitleFree('MyTitle')).be.true();
+		});
+
+		it('title is not free with text document sharing name', async () => {
+			await vscode.workspace.openTextDocument();
+			should(utils.isEditorTitleFree('Untitled-1')).be.false();
+		});
+
+		it('title is not free with notebook document sharing name', async () => {
+			await azdata.nb.showNotebookDocument(vscode.Uri.parse(`untitled:Untitled-1`));
+			should(utils.isEditorTitleFree('Untitled-1')).be.false();
+		});
+	});
+
 	describe('getClusterEndpoints', () => {
+		const baseServerInfo: azdata.ServerInfo = {
+			serverMajorVersion: -1,
+			serverMinorVersion: -1,
+			serverReleaseVersion: -1,
+			engineEditionId: -1,
+			serverVersion: '',
+			serverLevel: '',
+			serverEdition: '',
+			isCloud: false,
+			azureVersion: -1,
+			osVersion: '',
+			options: {}
+		};
+		it('empty endpoints does not error', () => {
+			const serverInfo = Object.assign({}, baseServerInfo);
+			serverInfo.options['clusterEndpoints'] = [];
+			should(utils.getClusterEndpoints(serverInfo).length).equal(0);
+		});
+
+		it('endpoints without endpoint field are created successfully', () => {
+			const serverInfo = Object.assign({}, baseServerInfo);
+			const ipAddress = 'localhost';
+			const port = '123';
+			serverInfo.options['clusterEndpoints'] = [{ ipAddress: ipAddress, port: port }];
+			const endpoints = utils.getClusterEndpoints(serverInfo);
+			should(endpoints.length).equal(1);
+			should(endpoints[0].endpoint).equal('https://localhost:123');
+		});
 
+		it('endpoints with endpoint field are created successfully', () => {
+			const endpoint = 'https://myActualEndpoint:8080';
+			const serverInfo = Object.assign({}, baseServerInfo);
+			serverInfo.options['clusterEndpoints'] = [{ endpoint: endpoint, ipAddress: 'localhost', port: '123' }];
+			const endpoints = utils.getClusterEndpoints(serverInfo);
+			should(endpoints.length).equal(1);
+			should(endpoints[0].endpoint).equal(endpoint);
+		});
+	});
+
+	describe('getHostAndPortFromEndpoint', () => {
+		it('valid endpoint is parsed correctly', () => {
+			const host = 'localhost';
+			const port = '123';
+			const hostAndIp = utils.getHostAndPortFromEndpoint(`https://${host}:${123}`);
```

Comment:
nit: for consistency may want to use ${port} instead of the literal here.

Reply:
hah, whoops. Yeah brain must have been on autopilot here.